### PR TITLE
Add compatibility matrix

### PR DESCRIPTION
**Pull request by CatherineF-dev**
[following metrics-server](https://github.com/kubernetes-sigs/metrics-server?tab=readme-ov-file#compatibility-matrix)

### CODE REVIEW

**Comment by dgrisonnet** [comment]
I wonder if it could be worth adding a collumn for compatibility with Kubernetes. Like for example the recent metrics-server releases are still compatible with Kubernetes 1.19+ evne though they are built on recent versions of the libraries: https://github.com/kubernetes-sigs/metrics-server?tab=readme-ov-file#compatibility-matrix

File: README.md [hunk 1]
```diff
@@ -32,6 +32,14 @@ found in the [test-adapter directory](/test-adapter).
 [Go](https://go.dev/doc/install): this library requires the same version of
 [Go as Kubernetes](https://git.k8s.io/community/contributors/devel/development.md#go).
 
+### Compatibility matrix
+
+| custom-metrics-apiserver |  Kubernetes Apiserver Version |
+|--------------------|:----------------------------:|
+| **v1.29.0**         | v1.29                      |
+| **v1.30.0**         | v1.30                      |
```
Comment (dgrisonnet) [inline README.md:40]: I think we should try to at least get all the supported Kubernetes versions in the compat matrix: https://kubernetes.io/releases/

Comment (CatherineF-dev) [reply README.md]: Updated. Followed metrics-server.

File: README.md [hunk 2]
```diff
@@ -32,6 +32,15 @@ found in the [test-adapter directory](/test-adapter).
 [Go](https://go.dev/doc/install): this library requires the same version of
 [Go as Kubernetes](https://git.k8s.io/community/contributors/devel/development.md#go).
 
+### Compatibility Matrix
```
Comment (rexagod) [inline README.md:35]: Needs to be updated.

**Review by rexagod — CHANGES_REQUESTED** [review]

**Pull request closed without merging**